packages/cli/src/environments.ee/source-control/source-control.controller.ee.ts

```diff
@@ -164,19 +166,16 @@ export class SourceControlController {
 	@Post('/push-workfolder', { middlewares: [sourceControlLicensedAndEnabledMiddleware] })
 	@GlobalScope('sourceControl:push')
 	async pushWorkfolder(
-		req: SourceControlRequest.PushWorkFolder,
+		req: AuthenticatedRequest,
 		res: express.Response,
+		@Body payload: PushWorkFolderRequestDto,
 	): Promise<SourceControlledFile[]> {
-		if (this.sourceControlPreferencesService.isBranchReadOnly()) {
-			throw new BadRequestError('Cannot push onto read-only branch.');
-		}
```

Comment:
Why was this removed?

Reply:
This is already done int he service, got it.

Reply:
not only that, but also if we were to add a public api endpoint for pushing, we'd have to repeat this same check there. leaving this check in the service seems like the correct thing to do, unless we want to implement this check in a middleware instead.